src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -2675,6 +2672,40 @@ def convert_func_to_agg_func_info(
         new_data_column_quoted_identifier = [
             col_agg_op.agg_snowflake_quoted_identifier for col_agg_op in agg_col_ops
         ]
+
+        # The ordering of the named aggregations is changed by us when we process
+        # the agg_kwargs into the func dict (named aggregations on the same
+        # column are moved to be contiguous, see groupby.py::aggregate for an
+        # example). We need to check if the order of the output columns is correct,
+        # and if not, reorder them.
+        if uses_named_aggs:
+            correct_ordering = list(agg_kwargs.keys())
```

Comment:
Are the vanity arguments `_formatted_named_kwargs` and `_correct_aggregation_order` not passed at this point?

Reply:
This codepath is in qc.groupby_agg - those vanity arguments are passed along the df/series.agg codepath (so they hit qc.agg) - so those arguments are not there here. (Those vanity arguments are actually passed as a surrogate to passing agg_kwargs as is done here)